tests/page-objects/reports/reports.po.js

```diff
@@ -3,8 +3,62 @@ const helper = require('../../helper');
 const itemSummary = '#reports-content .report-body .item-summary';
 const reportBody = '#reports-content .report-body';
 const reportBodyDetails = '#reports-content .report-body .details';
+const datePickerStart = element(by.css('.daterangepicker [name="daterangepicker_start"]'));
+const datePickerEnd = element(by.css('.daterangepicker [name="daterangepicker_end"]'));
+const dateFilter = element(by.css('#date-filter'));
+
+// Left hand side list elements
+const reportListID = '#reports-list';
+const list = element(by.css(reportListID));
+const listLoader =  element(by.css(`${reportListID} .loader`));
+const firstReport = element(by.css(`${reportListID} li:first-child`));
+const allReports = element.all(by.css(`${reportListID} li`));
+
+//Right hand side elements
+const reportSummary = element(by.css('#reports-content .item-summary')); 
+const submitterPhone = reportSummary.element(by.css('.sender .phone'));
+const submitterName =  reportSummary.element(by.css('.sender .name'));
+const subjectName = reportSummary.element(by.css('.subject .name'));
+const summaryFormName = reportSummary.element(by.css('.subject + div'));
+const formNameNoSubject = reportSummary.element(by.css('mm-sender + div'));
 
 module.exports = {
+  formNameNoSubject: formNameNoSubject,
+  subjectName: subjectName,
+  summaryFormName: summaryFormName,
+  submitterName: submitterName,
+  submitterPhone: submitterPhone,
+  allReports: allReports,
+  firstReport: firstReport,
+  listLoader: listLoader,
+  list: list,
+  subject: async reportElement =>  {
+    return reportElement.element(by.css('.content .heading h4 span'));
+  },
+  formName: async reportElement =>  {
+    return reportElement.element(by.css('.summary'));
+  },
+  loadReport: async uuid => {
+    const report = module.exports.reportByUUID(uuid).first();
+    await helper.waitElementToBeClickable(report);
+    await helper.clickElement(report);
+    await helper.waitElementToPresent(reportSummary, 3000);
+    return report;
+  },
+  reportByUUID: uuid => {
+    return list.all(by.css('li[data-record-id="' + uuid + '"]'));
+  },
+  filterByDate: (startDate, endDate) => {
+    let clear = '';
+    for (let i = 0; i < 20; i++) {
+      clear += protractor.Key.BACK_SPACE;
+    }
```

Comment:
I know you're just moving this to a better home, but while you're in here, would you please tidy up the syntax a bit?

This manual looping can be replaced with the following:
```suggestion
    const clear = protractor.Key.BACK_SPACE.repeat(20);
```
Nice to trim the line count down, but this is also easier to skim to get a good idea of what's going on without having to worry about reading the loop business.

Reply:
I removed the custom logic and sending backspace. Now using just the protractor function `clear()` on the element.